### ISSUE TITLE
Unknown `socialauth` users are not registered automatically upon first login

### ISSUE THREAD

**Issue by almereyda** [fiduswriter/fiduswriter#1193]
When logging in to an instance where I can prove legitimate access credentials through a third party, such as configured `socialauth` providers, we can imagine a Discourse-like workflow, where previously unknown users are suggested to register with the fields prefilled with the validated return values.

This way users receive the chance to actively consent to Terms of Use and Privacy Policy during sign up, while profiting of seamless integration of their account with an IAM system.

**Comment by almereyda** [comment]
This was discovered with unregistered people trying to login through socialauth, and being presented with an uncatched HTTP 500 error.

**Comment by johanneswilm** [comment]
Hey @almereyda, what were the settings for that server? As far as I am aware, the default configuration should not lead to uncaught 500 error messages if new people try to sign in using the social auth logins.

**Comment by almereyda** [comment]
I have found the associated error in the `ADMINS` inbox. It reads:

```py
Internal Server Error: /api/gitlab/gitlab/login/callback/

ProgrammingError at /api/gitlab/gitlab/login/callback/
operator does not exist: character varying = integer
LINE 1: ...e"."to_id" IS NULL AND ("user_userinvite"."email" IN (SELECT...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.


Request Method: GET
Request URL: https://writer.allmende.io/api/gitlab/gitlab/login/callback/?code=28d23c57b841b2da3a2c65d1c5560c6d6e3c712b92253ed12ee81c77f1b37532&state=YHqm6PoqgK4z
Django Version: 3.2.5
Python Executable: /usr/bin/python3
Python Version: 3.8.10
Python Path: ['/fiduswriter', '/usr/local/bin', '/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '/usr/local/lib/python3.8/dist-packages', '/usr/lib/python3/dist-packages', '/usr/local/lib/python3.8/dist-packages/fiduswriter']
Server time: Thu, 10 Feb 2022 08:23:55 +0000
Installed Applications:
['npm_mjs',
 'base',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.admin',
 'django.contrib.admindocs',
 'django.contrib.flatpages',
 'django_js_error_hook',
 'loginas',
 'fixturemedia',
 'browser_check',
 'menu',
 'document',
 'bibliography',
 'usermedia',
 'user',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'avatar',
 'feedback',
 'style',
 'allauth.socialaccount.providers.gitlab',
 'book',
 'citation_api_import',
 'languagetool',
 'ojs',
 'phplist']
Installed Middleware:
['django.middleware.common.CommonMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.locale.LocaleMiddleware']


Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)

The above exception (operator does not exist: character varying = integer
LINE 1: ...e"."to_id" IS NULL AND ("user_userinvite"."email" IN (SELECT...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
) was the direct cause of the following exception:
  File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/providers/oauth2/views.py", line 73, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/providers/oauth2/views.py", line 143, in dispatch
    return complete_social_login(request, login)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/helpers.py", line 152, in complete_social_login
    return _complete_social_login(request, sociallogin)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/helpers.py", line 174, in _complete_social_login
    ret = _process_signup(request, sociallogin)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/helpers.py", line 47, in _process_signup
    ret = complete_social_signup(request, sociallogin)
  File "/usr/local/lib/python3.8/dist-packages/allauth/socialaccount/helpers.py", line 179, in complete_social_signup
    return complete_signup(request,
  File "/usr/local/lib/python3.8/dist-packages/allauth/account/utils.py", line 184, in complete_signup
    signals.user_signed_up.send(sender=user.__class__,
  File "/usr/local/lib/python3.8/dist-packages/django/dispatch/dispatcher.py", line 180, in send
    return [
  File "/usr/local/lib/python3.8/dist-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "/usr/local/lib/python3.8/dist-packages/fiduswriter/user/signals.py", line 84, in on_user_signed_up
    invites_connect(sociallogin.account.user)
  File "/usr/local/lib/python3.8/dist-packages/fiduswriter/user/views.py", line 379, in invites_connect
    if len(invites) == 0:
  File "/usr/local/lib/python3.8/dist-packages/django/db/models/query.py", line 262, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.8/dist-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.8/dist-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/usr/local/lib/python3.8/dist-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/dist-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)

Exception Type: ProgrammingError at /api/gitlab/gitlab/login/callback/
Exception Value: operator does not exist: character varying = integer
LINE 1: ...e"."to_id" IS NULL AND ("user_userinvite"."email" IN (SELECT...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.

Request information:
USER: AnonymousUser

GET:
code = '28d23c57b841b2da3a2c65d1c5560c6d6e3c712b92253ed12ee81c77f1b37532'
state = 'YHqm6PoqgK4z'

POST: No POST data

FILES: No FILES data
```

and gives a hint:

> No operator matches the given name and argument types. You might need to add explicit type casts.

Here are the [Fiduswriter settings ($421) · Snippets](https://lab.allmende.io/-/snippets/421).

**Comment by johanneswilm** [comment]
This has been fixed in 3.11.4

**Comment by almereyda** [comment]
Many thanks! Will yield and update some time within the next month and
report back.
